Feature/create project

yardobr opened the pull request:
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23issuecomment-166676478%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23issuecomment-166677485%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23issuecomment-167164843%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23issuecomment-168499473%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23issuecomment-168499589%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23issuecomment-168501122%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23issuecomment-168503975%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23issuecomment-170700952%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48272252%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48272374%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48272765%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48273755%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48401192%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48681006%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48681013%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48681016%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48749694%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48754550%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48755519%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48755877%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48756080%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48976237%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48976277%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48976929%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48977181%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48978939%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48979035%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383803%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383804%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383805%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383807%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383808%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383810%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383812%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383815%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383818%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383820%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383978%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383979%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383980%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49383982%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r49384056%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23issuecomment-170835121%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23issuecomment-166676478%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22nice%21%5Cr%5CnI%27ve%20tried%20to%20create%20project%20and%20organization%20is%20created%5Cr%5Cnsome%20propositions%3A%5Cr%5Cn1.%20when%20I%20clicked%20%5C%22create%20project%5C%22%20in%20the%20modal%20window%20I%20was%20moved%20to%20trello%20Allow/Deny%20window%20and%20then%20came%20back%20and%20needed%20to%20click%20again%20%28it%20is%20not%20obvious%20workflow%29%5Cr%5Cn%5Cr%5Cn2.%20I%20don%27t%20see%20added%20project%20at%20the%20sidebar%20%3A%29%5Cr%5Cn%5Cr%5Cn3.%20I%20saw%20some%20code%20with%20animations%20of%20the%20right%20sidebar.%20can%20we%20close%20it%20somehow%3F%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-12-22T17%3A09%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22One%20more%21%5Cr%5CnSee%20Ann%27s%20task.%20She%20may%20do%20some%20tasks%20that%20overlaps%20with%20yours.%5Cr%5CnI%20propose%20you%20to%20focus%20on%20projects%20and%20%40annvaskovska%20to%20focus%20on%20log%20in%20%28for%20example%20we%20may%20need%20to%20display%20your%20nick%20on%20Trello%20somewhere%20at%20the%20top%20-%20added%20that%20to%20Ann%27s%20tasks%29.%22%2C%20%22created_at%22%3A%20%222015-12-22T17%3A12%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22can%20not%20be%20merged%20automatically%20%28komissar%20merge%20would%20not%20help%29%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-12-24T22%3A58%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22why%20there%20are%20so%20many%20files%20%2826%29%20from%20all%20over%20the%20project%3F%22%2C%20%22created_at%22%3A%20%222016-01-03T13%3A36%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22you%20need%20to%20clean%20this%20pull%20request.%20it%20seems%20that%20you%20have%20some%20old%20changes%20here%20from%20other%20team%20members%20that%20have%20overwritten%20the%20actual%20state%20of%20master%20branch%22%2C%20%22created_at%22%3A%20%222016-01-03T13%3A39%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22The%20amount%20of%20created%20and%20modified%20files%20is%2026%20because%20during%20the%20work%20I%27ve%20changed%20not%20only%20my%20pages%2C%20but%20the%20structure%20of%20the%20project%20and%20UI%20on%20different%20templates%20for%20sidebar%20to%20show%20properly.%5Cr%5CnBut%20after%20all%20UI%20is%20being%20changed%20not%20only%20by%20me%2C%20so%20now%20it%20doesn%27t%20look%20the%20way%20I%20thought.%5Cr%5CnHow%20can%20I%20clean%20my%20Pull%20request%20from%20all%20the%20old%20changes%3F%22%2C%20%22created_at%22%3A%20%222016-01-03T14%3A19%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20assume%20the%20best%20way%20is%20to%20manually%20remove%20from%20this%20pull%20request%20all%20files%20that%20are%20not%20in%20the%20scope%20of%20the%20%5C%22Feature/create%20project%5C%22%5Cr%5Cn%5Cr%5CnI%20hope%20you%20can%20manage%20that%20%3A%29%5Cr%5Cn%28with%20Webstorm%20it%20might%20be%20easier%2C%20but%20still%20no%20licence%29%22%2C%20%22created_at%22%3A%20%222016-01-03T14%3A53%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%2C%20%7B%22body%22%3A%20%22we%20can%20merge%20your%20commit.%20After%20squashing%20everything%20in%20one%20commit.%20The%20only%20thing%20that%20I%27d%20better%20change%20is%20the%20naming%20of%20modals%3A%20%60modal%60%20and%20%60newModal%60%20to%20smt.%20like%20%60createProjectModal%60%20and%20%60projectCreatedModal%60%22%2C%20%22created_at%22%3A%20%222016-01-12T08%3A21%3A41Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2004a25a0aa84af5da983e4b9e1d07b6a389d66ada%20TCMSApp/src/client/app/core/config.js%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48272252%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60Trello%60%20is%20something%20exposed%20by%20https%3A//api.trello.com/1/client.js%3Fkey%3D50c269d155e10e2d93a95a16c3b2d9b3%20%3F%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-12-22T16%3A46%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/core/config.js%3AL18-25%22%7D%2C%20%22Pull%20e8743a1ae2cea7d78205a3a36906c47ddf57b8eb%20TCMSApp/src/client/app/admin/team.html%20474%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48681006%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20camel%20case%20in%20Id%27s%3F%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-01-01T23%3A06%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/admin/team.html%3AL1-343%22%7D%2C%20%22Pull%20e8743a1ae2cea7d78205a3a36906c47ddf57b8eb%20TCMSApp/src/client/app/apis/trello/auth.servise.js%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48681013%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22oh%2C%20filename%20again%20%3A%29%20SERVISE%20is%20wrong%20%22%2C%20%22created_at%22%3A%20%222016-01-01T23%3A08%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/apis/trello/auth.servise.js%3AL1-32%22%7D%2C%20%22Pull%2066cf0af15f0112464102843c657f85887fec8b2e%20TCMSApp/src/client/app/apis/trello/create-project.service.js%2037%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48755877%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22you%20are%20pushing%20modal%20tooooo%20deep.%20this%20logic%20%2Amust%2A%20be%20in%20controller%20%28and%20then%20you%20should%20not%20pass%20here%20vm%20and%20modal%29%22%2C%20%22created_at%22%3A%20%222016-01-04T17%3A29%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A48%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/apis/trello/create-project.service.js%3AL1-56%22%7D%2C%20%22Pull%204dba4a8b3362d4c2f0ebcc662caf70531f441b1f%20TCMSApp/src/client/app/apis/trello/create-project.service.js%2047%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48976237%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22one-line%20code%20will%20look%20like%3A%5Cr%5Cn%60%60%60js%5Cr%5Cnreturn%20%28res.filter%28function%28e%29%20%7B%20return%20e.displayName%20%3D%3D%3D%20projectName%3B%20%7D%29.length%20%3E%200%29%5Cr%5Cn%60%60%60%22%2C%20%22created_at%22%3A%20%222016-01-06T16%3A27%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22but%20your%20is%20ok%20too%20%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222016-01-06T16%3A27%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/apis/trello/create-project.service.js%3AL1-70%22%7D%2C%20%22Pull%20e8743a1ae2cea7d78205a3a36906c47ddf57b8eb%20TCMSApp/src/client/app/defects/defects.html%203%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48681016%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22hm....%20unmerged%20file%3F%20clear%20this%22%2C%20%22created_at%22%3A%20%222016-01-01T23%3A09%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/defects/defects.html%3AL1-11%22%7D%2C%20%22Pull%204dba4a8b3362d4c2f0ebcc662caf70531f441b1f%20TCMSApp/src/client/app/layout/sidebar.controller.js%2022%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48978939%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22do%20the%20%60vm.changeModal%60%20need%20to%20live%20in%20the%20scope%20of%20%60SidebarController%60%3F%5Cr%5CnIf%20we%20use%20it%20only%20inside%20the%20main%20modal%20it%20can%20live%20there%20maybe.%22%2C%20%22created_at%22%3A%20%222016-01-06T16%3A48%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A48%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/layout/sidebar.controller.js%3AL1-77%22%7D%2C%20%22Pull%2004a25a0aa84af5da983e4b9e1d07b6a389d66ada%20TCMSApp/src/client/index.html%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48273755%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22and%20of%20course%20we%20need%20to%20invent%20the%20way%20not%20to%20store%20key%20here%20%28ideally%20-%20so%20that%20every%20project%20creator%20can%20get%20own%20key%29%5Cr%5Cn%5Cr%5Cnbut%20it%20is%20a%20new%20task%20%28you%20can%20create%20it%29%22%2C%20%22created_at%22%3A%20%222015-12-22T16%3A59%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/index.html%3AL39-49%22%7D%2C%20%22Pull%2066cf0af15f0112464102843c657f85887fec8b2e%20TCMSApp/src/client/app/layout/sidebar.html%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48749694%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22still%20vm%20syntax%22%2C%20%22created_at%22%3A%20%222016-01-04T16%3A36%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A48%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/layout/sidebar.html%3AL4-22%22%7D%2C%20%22Pull%2066cf0af15f0112464102843c657f85887fec8b2e%20TCMSApp/src/client/app/apis/trello/create-project.service.js%2029%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48755519%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22lines%2020-28%20%28and%20%60%20logger.error%28%27Project%20with%20the%20same%20name%20already%20exists.%27%2C%20%27%27%2C%20%27Error%27%29%3B%60%29%20could%20be%20the%20separate%20function%20called%20%60noSuchProject%60%20returning%20true%20if%20no%20such%20project%20and%20logging%20error%20if%20there%20is%20and%20called%20as%3A%5Cr%5Cn%5Cr%5Cn%60%60%60js%5Cr%5Cnif%20%28noSuchProject%28%29%29%20%7B%5Cr%5Cn%20%20%20Trello.post%28%27/organizations%27%2C%20%7B%5Cr%5Cn%20%20%20%20%20%20...%5Cr%5Cn%7D%5Cr%5Cn%60%60%60%22%2C%20%22created_at%22%3A%20%222016-01-04T17%3A26%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/apis/trello/create-project.service.js%3AL1-56%22%7D%2C%20%22Pull%2004a25a0aa84af5da983e4b9e1d07b6a389d66ada%20TCMSApp/src/client/app/layout/layout.config.js%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48272374%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22do%20we%20need%20this%20empty%20file%3F%22%2C%20%22created_at%22%3A%20%222015-12-22T16%3A47%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/layout/layout.config.js%3AL1-7%22%7D%2C%20%22Pull%2066cf0af15f0112464102843c657f85887fec8b2e%20TCMSApp/src/client/app/layout/sidebar.controller.js%2082%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48756080%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22do%20we%20use%20getNavRoutes%3F%3F%22%2C%20%22created_at%22%3A%20%222016-01-04T17%3A31%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A48%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/layout/sidebar.controller.js%3AL1-85%22%7D%2C%20%22Pull%2004a25a0aa84af5da983e4b9e1d07b6a389d66ada%20TCMSApp/src/client/app/layout/sidebar.controller.js%2030%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48401192%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22lets%20be%20consistent.%20%60vmSidebarModal%60%20or%20something%20like%20that%22%2C%20%22created_at%22%3A%20%222015-12-24T07%3A55%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22Really%2C%20why%20%60sc%60%3F%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-01-06T16%3A33%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/layout/sidebar.controller.js%3AL1-93%22%7D%2C%20%22Pull%204dba4a8b3362d4c2f0ebcc662caf70531f441b1f%20TCMSApp/src/client/app/layout/sidebar.controller.js%2085%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48977181%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22why%20do%20you%20need%20%60states%60%3F%22%2C%20%22created_at%22%3A%20%222016-01-06T16%3A35%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A48%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/layout/sidebar.controller.js%3AL1-77%22%7D%2C%20%22Pull%2004a25a0aa84af5da983e4b9e1d07b6a389d66ada%20TCMSApp/src/client/app/layout/add-project.html%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48272765%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22no%20one%20likes%20inline%20styles%5Cr%5Cnespecially%20if%20you%27re%20doing%20margin-top%3A10px%20at%2010%20DIFFERENT%20places.%5Cr%5Cnit%20can%20be%20done%20better%29%20%22%2C%20%22created_at%22%3A%20%222015-12-22T16%3A50%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/layout/add-project.html%3AL1-23%22%7D%2C%20%22Pull%2066cf0af15f0112464102843c657f85887fec8b2e%20TCMSApp/src/client/app/layout/sidebar.service.js%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-012/pull/86%23discussion_r48754550%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22if%20you%20call%20the%20file%20sidebar.service%20it%20should%20be%20one%20service%20called%20sidebar.%20but%20here%20we%20have%20two%20factories.%20%5Cr%5Cnlet%20it%20be%20one%20returning%20functions%3A%5Cr%5Cn-%20addProject%5Cr%5Cn-%20fetchProjects%20%28intentionally%20had%20not%20used%20verb%20GET%20because%20we%20shall%20not%20do%20http%20GET%20here.%29%22%2C%20%22created_at%22%3A%20%222016-01-04T17%3A17%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22styleguide%20and%20experts%20teach%20us%20too%20write%20as%20follows%3A%20%5Cr%5Cn%5Cr%5Cn%60%60%60js%20%20%20%20%5Cr%5Cnfunction%20sidebarFactory%28moment%29%20%7B%5Cr%5Cn%20%20%20%20%20%20%20%20return%20%7B%5Cr%5Cn%20%20%20%20%20%20%20%20%20%20%20%20findProjectsNames%3A%20addProject%2C%5Cr%5Cn%20%20%20%20%20%20%20%20%20%20%20%20addProject%3A%20addProject%5Cr%5Cn%20%20%20%20%20%20%20%20%7D%5Cr%5Cn%20%20%20%20%7D%5Cr%5Cn%60%60%60%5Cr%5Cnand%20declare%20functions%20at%20the%20bottom%22%2C%20%22created_at%22%3A%20%222016-01-06T16%3A48%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7273003%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-01-11T21%3A47%3A19Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/16047940%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/yardobr%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20TCMSApp/src/client/app/layout/sidebar.service.js%3AL1-47%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>
- [ ] <a href='#crh-comment-General Comment'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#issuecomment-166676478'>General Comment</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> nice!
I've tried to create project and organization is created
some propositions:
1. when I clicked "create project" in the modal window I was moved to trello Allow/Deny window and then came back and needed to click again (it is not obvious workflow)
2. I don't see added project at the sidebar :)
3. I saw some code with animations of the right sidebar. can we close it somehow? :)
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> One more!
See Ann's task. She may do some tasks that overlaps with yours.
I propose you to focus on projects and @annvaskovska to focus on log in (for example we may need to display your nick on Trello somewhere at the top - added that to Ann's tasks).
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> can not be merged automatically (komissar merge would not help) :)
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> why there are so many files (26) from all over the project?
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> you need to clean this pull request. it seems that you have some old changes here from other team members that have overwritten the actual state of master branch
- <a href='https://github.com/yardobr'><img border=0 src='https://avatars.githubusercontent.com/u/16047940?v=3' height=16 width=16'></a> The amount of created and modified files is 26 because during the work I've changed not only my pages, but the structure of the project and UI on different templates for sidebar to show properly.
But after all UI is being changed not only by me, so now it doesn't look the way I thought.
How can I clean my Pull request from all the old changes?
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> I assume the best way is to manually remove from this pull request all files that are not in the scope of the "Feature/create project"
I hope you can manage that :)
(with Webstorm it might be easier, but still no licence)
- <a href='https://github.com/yardobr'><img border=0 src='https://avatars.githubusercontent.com/u/16047940?v=3' height=16 width=16'></a> :+1:<a href='#crh-update-none'></a>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> we can merge your commit. After squashing everything in one commit. The only thing that I'd better change is the naming of modals: `modal` and `newModal` to smt. like `createProjectModal` and `projectCreatedModal`
- [x] <a href='#crh-comment-Pull 04a25a0aa84af5da983e4b9e1d07b6a389d66ada TCMSApp/src/client/app/core/config.js 4'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48272252'>File: TCMSApp/src/client/app/core/config.js:L18-25</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> `Trello` is something exposed by https://api.trello.com/1/client.js?key=50c269d155e10e2d93a95a16c3b2d9b3 ?
- [x] <a href='#crh-comment-Pull 04a25a0aa84af5da983e4b9e1d07b6a389d66ada TCMSApp/src/client/app/layout/layout.config.js 1'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48272374'>File: TCMSApp/src/client/app/layout/layout.config.js:L1-7</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> do we need this empty file?
- [x] <a href='#crh-comment-Pull 04a25a0aa84af5da983e4b9e1d07b6a389d66ada TCMSApp/src/client/app/layout/add-project.html 15'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48272765'>File: TCMSApp/src/client/app/layout/add-project.html:L1-23</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> no one likes inline styles
especially if you're doing margin-top:10px at 10 DIFFERENT places.
it can be done better)
- [x] <a href='#crh-comment-Pull 04a25a0aa84af5da983e4b9e1d07b6a389d66ada TCMSApp/src/client/index.html 6'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48273755'>File: TCMSApp/src/client/index.html:L39-49</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> and of course we need to invent the way not to store key here (ideally - so that every project creator can get own key)
but it is a new task (you can create it)
- [x] <a href='#crh-comment-Pull 04a25a0aa84af5da983e4b9e1d07b6a389d66ada TCMSApp/src/client/app/layout/sidebar.controller.js 30'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48401192'>File: TCMSApp/src/client/app/layout/sidebar.controller.js:L1-93</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> lets be consistent. `vmSidebarModal` or something like that
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> Really, why `sc`? :)
- [x] <a href='#crh-comment-Pull e8743a1ae2cea7d78205a3a36906c47ddf57b8eb TCMSApp/src/client/app/admin/team.html 474'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48681006'>File: TCMSApp/src/client/app/admin/team.html:L1-343</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> why camel case in Id's? :)
- [x] <a href='#crh-comment-Pull e8743a1ae2cea7d78205a3a36906c47ddf57b8eb TCMSApp/src/client/app/apis/trello/auth.servise.js 1'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48681013'>File: TCMSApp/src/client/app/apis/trello/auth.servise.js:L1-32</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> oh, filename again :) SERVISE is wrong
- [x] <a href='#crh-comment-Pull e8743a1ae2cea7d78205a3a36906c47ddf57b8eb TCMSApp/src/client/app/defects/defects.html 3'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48681016'>File: TCMSApp/src/client/app/defects/defects.html:L1-11</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> hm.... unmerged file? clear this
- [x] <a href='#crh-comment-Pull 66cf0af15f0112464102843c657f85887fec8b2e TCMSApp/src/client/app/layout/sidebar.html 15'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48749694'>File: TCMSApp/src/client/app/layout/sidebar.html:L4-22</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> still vm syntax
- [x] <a href='#crh-comment-Pull 66cf0af15f0112464102843c657f85887fec8b2e TCMSApp/src/client/app/layout/sidebar.service.js 1'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48754550'>File: TCMSApp/src/client/app/layout/sidebar.service.js:L1-47</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> if you call the file sidebar.service it should be one service called sidebar. but here we have two factories.
let it be one returning functions:
- addProject
- fetchProjects (intentionally had not used verb GET because we shall not do http GET here.)
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> styleguide and experts teach us too write as follows:
```js
function sidebarFactory(moment) {
return {
findProjectsNames: addProject,
addProject: addProject
}
}
```
and declare functions at the bottom
- [x] <a href='#crh-comment-Pull 66cf0af15f0112464102843c657f85887fec8b2e TCMSApp/src/client/app/apis/trello/create-project.service.js 29'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48755519'>File: TCMSApp/src/client/app/apis/trello/create-project.service.js:L1-56</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> lines 20-28 (and ` logger.error('Project with the same name already exists.', '', 'Error');`) could be the separate function called `noSuchProject` returning true if no such project and logging error if there is and called as:
```js
if (noSuchProject()) {
Trello.post('/organizations', {
...
}
```
- [x] <a href='#crh-comment-Pull 66cf0af15f0112464102843c657f85887fec8b2e TCMSApp/src/client/app/apis/trello/create-project.service.js 37'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48755877'>File: TCMSApp/src/client/app/apis/trello/create-project.service.js:L1-56</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> you are pushing modal tooooo deep. this logic *must* be in controller (and then you should not pass here vm and modal)
- [x] <a href='#crh-comment-Pull 66cf0af15f0112464102843c657f85887fec8b2e TCMSApp/src/client/app/layout/sidebar.controller.js 82'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48756080'>File: TCMSApp/src/client/app/layout/sidebar.controller.js:L1-85</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> do we use getNavRoutes??
- [x] <a href='#crh-comment-Pull 4dba4a8b3362d4c2f0ebcc662caf70531f441b1f TCMSApp/src/client/app/apis/trello/create-project.service.js 47'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48976237'>File: TCMSApp/src/client/app/apis/trello/create-project.service.js:L1-70</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> one-line code will look like:
```js
return (res.filter(function(e) { return e.displayName === projectName; }).length > 0)
```
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> but your is ok too :+1:
- [x] <a href='#crh-comment-Pull 4dba4a8b3362d4c2f0ebcc662caf70531f441b1f TCMSApp/src/client/app/layout/sidebar.controller.js 85'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48977181'>File: TCMSApp/src/client/app/layout/sidebar.controller.js:L1-77</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> why do you need `states`?
- [x] <a href='#crh-comment-Pull 4dba4a8b3362d4c2f0ebcc662caf70531f441b1f TCMSApp/src/client/app/layout/sidebar.controller.js 22'></a> <img src='http://www.codereviewhub.com/site/github-completed.png' height=16 width=60>&nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-012/pull/86#discussion_r48978939'>File: TCMSApp/src/client/app/layout/sidebar.controller.js:L1-77</a></b>
- <a href='https://github.com/ITsvetkoFF'><img border=0 src='https://avatars.githubusercontent.com/u/7273003?v=3' height=16 width=16'></a> do the `vm.changeModal` need to live in the scope of `SidebarController`?
If we use it only inside the main modal it can live there maybe.


<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-012/pull/86?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-012/pull/86?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/ITsvetkoFF/Kv-012/pull/86'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

ITsvetkoFF commented on TCMSApp/src/client/app/core/config.js:
```diff
@@ -18,6 +18,7 @@
 
         delete window.moment;
         delete window.faker;
+        delete window.Trello;
```
`Trello` is something exposed by https://api.trello.com/1/client.js?key=50c269d155e10e2d93a95a16c3b2d9b3 ?

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented on TCMSApp/src/client/app/layout/layout.config.js:
```diff
@@ -0,0 +1,6 @@
+(function () {
```
do we need this empty file?

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented on TCMSApp/src/client/app/layout/add-project.html:
```diff
@@ -0,0 +1,22 @@
+                <textarea name="pr-descr" id="pr-descr" rows="3" class="col-md-12" style="resize: none;" required ng-init="vmChild.projectDescription=''" ng-model="vmChild.projectDescription"></textarea>
+            </div>
+
+            <div class="container-fluid" style="margin-top: 10px; margin-bottom: 15px;">
```
no one likes inline styles
especially if you're doing margin-top:10px at 3 places.
it can be done better)

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented on TCMSApp/src/client/index.html:
```diff
@@ -39,6 +39,10 @@
 
+
+    <script src="//code.jquery.com/jquery-1.11.3.min.js"></script>
+    <script src="https://api.trello.com/1/client.js?key=50c269d155e10e2d93a95a16c3b2d9b3"></script>
```
and of course we need to invent the way not to store key here (ideally - so that every project creator can get own key)

but it is a new task (you can create it)

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented:
nice!
I've tried to create project and organization is created
some propositions:
1. when I clicked "create project" in the modal window I was moved to trello Allow/Deny window and then came back and needed to click again (it is not obvious workflow)

2. I don't see added project at the sidebar :)

3. I saw some code with animations of the right sidebar. can we close it somehow? :)

ITsvetkoFF commented:
One more!
See Ann's task. She may do some tasks that overlaps with yours.
I propose you to focus on projects and @annvaskovska to focus on log in (for example we may need to display your nick on Trello somewhere at the top - added that to Ann's tasks).

ITsvetkoFF commented on TCMSApp/src/client/app/layout/sidebar.controller.js:
```diff
@@ -1,24 +1,92 @@
+                    };
+
+                },
+                controllerAs: 'sc'
```
lets be consistent. `sidebarVm`

ITsvetkoFF replied:
Really, why sc? :)

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented:
can not be merged automatically (komissar merge would not help) :)

ITsvetkoFF commented on TCMSApp/src/client/app/admin/team.html:
```diff
@@ -1,340 +1,342 @@
 
-<div id="addPerson" class="modal fade" role="dialog">
-    <div class="modal-dialog">
+    <div id="addPerson" class="modal fade" role="dialog">
```
why camel case in Id's? :)

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented on TCMSApp/src/client/app/apis/trello/auth.servise.js:
```diff
@@ -0,0 +1,31 @@
+(function () {
```
oh, filename again :) SERVISE is wrong

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented on TCMSApp/src/client/app/defects/defects.html:
```diff
@@ -1,6 +1,10 @@
 <section class="mainbar">
     <div class="matter">
+<<<<<<< HEAD
```
hm.... unmerged file? clear this

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented:
why there are so many files (26) from all over the project?

ITsvetkoFF commented:
you need to clean this pull request. it seems that you have some old changes here from other team members that have overwritten the actual state of master branch

yardobr commented:
The amount of created and modified files is 26 because during the work I've changed not only my pages, but the structure of the project and UI on different templates for sidebar to show properly.
But after all UI is being changed not only by me, so now it doesn't look the way I thought.
How can I clean my Pull request from all the old changes?

ITsvetkoFF commented:
I assume the best way is to manually remove from this pull request all files that are not in the scope of the "Feature/create project"

I hope you can manage that :)
(with Webstorm it might be easier, but still no licence)

ITsvetkoFF commented on TCMSApp/src/client/app/layout/sidebar.html:
```diff
@@ -4,19 +4,18 @@
-                </li>
-                <li class="nlightblue fade-selection-animation">
-                    <a> Project 4</a>
+                <li class="nlightblue fade-selection-animation" ng-repeat="projectName in vm.projectsNames">
```
still vm syntax

yardobr replied:
:+1:

ITsvetkoFF commented on TCMSApp/src/client/app/layout/sidebar.service.js:
```diff
@@ -0,0 +1,46 @@
+(function () {
```
if you call the file sidebar.service it should be one service called sidebar. but here we have two factories. 
let it be one returning functions:
- addProject
- fetchProjects (intentionally had not used verb GET because we shall not do http GET here.)

ITsvetkoFF replied:
styleguide and experts teach us too write as follows: 

```js    
function sidebarFactory(moment) {
        return {
            findProjectsNames: addProject,
            addProject: addProject
        }
    }
```
and declare functions at the bottom

yardobr replied:
:+1:<a href='#crh-update-complete'></a>

ITsvetkoFF commented on TCMSApp/src/client/app/apis/trello/create-project.service.js:
```diff
@@ -0,0 +1,55 @@
+                            break;
+                        }
+                    }
+                    if (same) logger.error('Project with the same name already exists.', '', 'Error');
```
lines 20-28 (and ` logger.error('Project with the same name already exists.', '', 'Error');`) could be the separate function called `noSuchProject` returning true if no such project and logging error if there is and called as:

```js
if (noSuchProject()) {
   Trello.post('/organizations', {
      ...
}
```

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented on TCMSApp/src/client/app/apis/trello/create-project.service.js:
```diff
@@ -0,0 +1,55 @@
+                            desc: projectDescription
+                        }).then(function (res) {
+                                logger.success('Project ' + projectName + ' created. Description: ' + projectDescription, '', 'Project created');
+                                vm.changeModal(modalWindow);
```
you are pushing modal tooooo deep. this logic must be in controller (and then you should not pass here vm and modal)

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented on TCMSApp/src/client/app/layout/sidebar.controller.js:
```diff
@@ -1,28 +1,84 @@
+            getNavRoutes();
+        }
 
         function getNavRoutes() {
```
do we use getNavRoutes??

yardobr replied:
:+1:<a href='#crh-update-complete'></a>

ITsvetkoFF commented on TCMSApp/src/client/app/apis/trello/create-project.service.js:
```diff
@@ -0,0 +1,69 @@
+                }
+            );
+
+            function sameProjectExists(res, projectName) {
```
one-line code will look like:
```js
return (res.filter(function(e) { return e.displayName === projectName; }).length > 0)
```

ITsvetkoFF replied:
but your is ok too :+1:

ITsvetkoFF commented on TCMSApp/src/client/app/layout/sidebar.controller.js:
```diff
@@ -1,28 +1,76 @@
+            );
+        };
+
+        var states = routerHelper.getStates();
```
why do you need `states`?

yardobr replied:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented on TCMSApp/src/client/app/layout/sidebar.controller.js:
```diff
@@ -1,28 +1,76 @@
-        activate();
+        vm.projectsNames = sidebarFactory.findProjectsNames();
+
+        vm.changeModal = function (modal) {
```
do the `vm.changeModal` need to live in the scope of `SidebarController`?
If we use it only inside the main modal it can live there maybe.

yardobr replied:
:+1:<a href='#crh-update-none'></a>

yardobr commented:
:+1:<a href='#crh-update-none'></a>

ITsvetkoFF commented:
we can merge your commit. After squashing everything in one commit. The only thing that I'd better change is the naming of modals: `modal` and `newModal` to smt. like `createProjectModal` and `projectCreatedModal`
